server/databases_test.go

```diff
@@ -308,6 +315,248 @@ func TestService_DropRetentionPolicy(t *testing.T) {
 	}
 }
 
+func TestService_Measurements(t *testing.T) {
+	type fields struct {
+		SourcesStore chronograf.SourcesStore
+		Logger       chronograf.Logger
+		Databases    chronograf.Databases
+	}
+	type args struct {
+		queryParams map[string]string
+	}
+	type wants struct {
+		statusCode int
+		body       string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		wants  wants
+	}{
+		{
+			name: "Gets 100 measurements when no limit or offset provided",
+			fields: fields{
+				SourcesStore: &mocks.SourcesStore{
+					GetF: func(ctx context.Context, srcID int) (chronograf.Source, error) {
+						return chronograf.Source{
+							ID: 0,
+						}, nil
+					},
+				},
+				Databases: &mocks.Databases{
+					ConnectF: func(context.Context, *chronograf.Source) error {
+						return nil
+					},
+					GetMeasurementsF: func(ctx context.Context, dbID string, limit, offset int) ([]chronograf.Measurement, error) {
+						return []chronograf.Measurement{
+							{
+								Name: "pineapple",
+							},
+							{
+								Name: "cubeapple",
+							},
+							{
+								Name: "pinecube",
+							},
+						}, nil
+					},
+				},
+			},
+			args: args{
+				queryParams: map[string]string{},
+			},
+			wants: wants{
+				statusCode: 200,
+				body: `{"measurements":[{"name":"pineapple"},{"name":"cubeapple"},{"name":"pinecube"}],"links":{"self":"/chronograf/v1/sources/0/dbs/0/measurements?limit=100\u0026offset=0","first":"/chronograf/v1/sources/0/dbs/0/measurements?limit=100\u0026offset=0","next":"/chronograf/v1/sources/0/dbs/0/measurements?limit=100\u0026offset=100"}}
```

Comment:
Weird that the ampersand is unicode here.

Reply:
i thought the same. see here for explanation: https://golang.org/pkg/encoding/json/#Marshal:
> String values encode as JSON strings coerced to valid UTF-8, replacing invalid bytes with the Unicode replacement rune. The angle brackets "<" and ">" are escaped to "\u003c" and "\u003e" to keep some browsers from misinterpreting JSON output as HTML. Ampersand "&" is also escaped to "\u0026" for the same reason. This escaping can be disabled using an Encoder that had SetEscapeHTML(false) called on it.